meson.build

```diff
@@ -0,0 +1,151 @@
+project('pyfmmlib', 'c', 'fortran',
+  version : '2023.1',
+  license: 'MIT',
+  meson_version: '>=0.64.0',
+  default_options : ['warning_level=2', 'fortran_std=legacy'],
+)
+
+# {{{ gather includes
+
+py_mod = import('python')
+py = py_mod.find_installation(pure: false)
+py_dep = py.dependency()
+
+# NOTE: meson does not like absolute paths, so we dance around it
+incdir_numpy = run_command(py,
+  [
+    '-c',
+    '''import os
+import numpy as np
+try:
+  incdir = os.path.relpath(np.get_include())
+except Exception:
+  incdir = np.get_include()
+print(incdir)
+  '''], check: true).stdout().strip()
+inc_numpy = include_directories(incdir_numpy)
+
+incdir_f2py = incdir_numpy / '..' / '..' / 'f2py' / 'src'
+inc_f2py = include_directories(incdir_f2py)
+
+# }}}
+
+# {{{ generate vec wrappers
+
+generate_wrappers = files('vec_wrappers.py')
+wrapper_vec = custom_target('wrapper_vec',
+  output: 'vec_wrappers.f90',
+  command: [py, generate_wrappers, '@OUTPUT@'])
+
+# }}}
+
+# {{{ generate f2py wrappers
+
+f2py_name = '_internal'
+f2py_module_c = f'@f2py_name@module.c'
+f2py_wrapper_f = f'@f2py_name@-f2pywrappers.f'
+
+generate_f2py = files('generate-f2py.py')
+f2py_dep = custom_target('f2py',
+  input: 'wrappers.pyf.mako',
+  output: [f2py_module_c, f2py_wrapper_f],
+  command: [py, generate_f2py, '@INPUT@', '-o', '@OUTDIR@'])
+
+# }}}
+
+
+# {{{ fortranobject
+
+f2py_object_c = 'fortranobject.c'
+f2py_object_static = static_library('_fortranobject',
+  incdir_f2py / f2py_object_c,
+  dependencies: py_dep,
+  include_directories: [inc_numpy, inc_f2py])
+
+f2py_object_dep = declare_dependency(
+  link_with: f2py_object_static,
+  include_directories: [inc_numpy, inc_f2py])
+
+# }}}
+
+# {{{ fmmlib
+
+fmmlib2d_sources = [
+  'fmmlib2d/src/cdjseval2d.f',
+  'fmmlib2d/src/d2mtreeplot.f',
```

Comment:
```suggestion
```

Reply:
Removed here and in the 3d sources too:
https://github.com/inducer/pyfmmlib/compare/940dd0a659198eb9113e0cd0db975b88bb969dca..f0a48f5676ee4954828ff306ff6b8aca834b1b9a